clang/include/clang/Driver/Options.td

```diff
@@ -3268,8 +3268,9 @@ defm stack_arrays : BooleanFFlag<"stack-arrays">, Group<gfortran_Group>;
 defm underscoring : BooleanFFlag<"underscoring">, Group<gfortran_Group>;
 defm whole_file : BooleanFFlag<"whole-file">, Group<gfortran_Group>;
 
-def sycl : Flag<["--"], "sycl">,
+def sycl_device_only : Flag<["-"], "fsycl-device-only">,
```

Comment:
I suggest using `CC1Option` to simplify maintenance and remove similar definition in `CC1Options.td`.
```
def sycl_device_only : Flag<["-"], "fsycl-device-only">, Flags<[CC1Option]>,
```

Reply:
I believe making that change would also require updating a large number of tests (80+).  If you think it is worthwhile then perhaps it would be better to do that in a subsequent change?

Reply:
> I believe making that change would also require updating a large number of tests (80+).

Why do you think so?
It looks like CC1 option has exactly the same name, so I'm not sure why this change would require updating any tests.

```
def fsycl_is_device : Flag<["-"], "fsycl-is-device">,
  HelpText<"Generate code for SYCL device.">;
```

This might require updates in the source code, but this should be an easy find/replace action.
I'm okay to apply this change in a separate PR.

Reply:
Maybe I'm misunderstanding, but it appears to me one is "sycl_device_only" while the other is "fsycl_is_device".  This model follows the other exisiting "device_only" option in Options.td.

Reply:
> Maybe I'm misunderstanding, but it appears to me one is "sycl_device_only" while the other is "fsycl_is_device". This model follows the other exisiting "device_only" option in Options.td.

These are the names of clang options used in **clang's source code**. Tests are using `-fsycl-is-device`/`--sycl`, so I don't think tests changes are needed. I expect only clang's source code changes.
Does it make sense?